main.lua

```diff
@@ -0,0 +1,72 @@
+local Point = {}
+Point.__index = Point
+Point.__lt = function(a, b)
+  if a.x == b.x then
+    return a.y < b.y
+  end
+  return a.x < b.x
+end
+
+function Point:new(x, y, idx)
+  return setmetatable({ x = x, y = y, idx = idx }, Point)
+end
+
+while true do
+  local ranks = {}
+
+  function find_ranks(...)
```

Comment:
Why not just pass `pts` but `table.unpack(pts)` with `...`?

Reply:
It is bad idea to use `...` ,and I will change it.